src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -184,6 +189,54 @@ public AzureStorageAccountResource getOrCreateStorageAccount(
         storageAccountName, applicationResource, region, flightId);
   }
 
+  public void createSnapshotStorageAccount(
```

Comment:
I'm surprised there is need for a completely separate method from the getOrCreateDatasetStorage account. Is there more code that could be shared?

Reply:
I took a look, and because the snapshots have only one billing project while datasets have many, the logic is different enough that I think combining them would end up being more complicated. I've simplified the logic in both cases, so hopefully its a bit clearer.